main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -911,13 +911,14 @@ object Weeder2 {
     private def visitQnameExpr(tree: Tree)(implicit sctx: SharedContext): Expr.Ambiguous = {
       expect(tree, TreeKind.QName)
       val idents = pickAll(TreeKind.Ident, tree).map(tokenToIdent)
+      val trailingDot = tryPick(TreeKind.TrailingDot, tree).nonEmpty
       assert(idents.nonEmpty) // Require at least one ident
       val first = idents.head
       val ident = idents.last
       val nnameIdents = idents.dropRight(1)
       val loc = SourceLocation(isReal = true, first.loc.sp1, ident.loc.sp2)
       val nname = Name.NName(nnameIdents, loc)
-      val qname = Name.QName(nname, ident, loc)
+      val qname = Name.QName(nname, ident, loc, trailingDot)
```

Comment:
As you were saying, we could also here just produce `["A", "B"], ""` it feels like that would make the other code more consistent (i.e. not have the boolean on the qname)